docs/attachapi.md

```diff
@@ -0,0 +1,99 @@
+
+```
+~/tmp $ ls -al
+total 0
+drwxr-xr-x   3 user_a staff    96  6 Aug 17:11 .
+drwxr-xr-x+ 89 user_a staff  2848  6 Aug 17:11 ..
+drwxrwxrwx+  7 root   staff   224  6 Aug 17:22 .com_ibm_tools_attach
+```
+
+In the default Attach API directory you can find certain files that start with an underscore `_*`, which are involved in synchronization.
+These files can be owned by any user but must have read and write permissions set. The files are empty and are automatically re-created if deleted.
+When your application attaches to a VM, a process directory is created.
+
+```
+~/tmp/.com_ibm_tools_attach $ ls -l
+total 3
+-rw-rw-rw-  1 user_a  staff    0  6 Aug 17:12 _attach_lock
+-rw-rw-rw-  1 user_a  staff    0  6 Aug 17:12 _master
+-rw-rw-rw-  1 user_a  staff    0  6 Aug 17:12 _notifier
+drwx--x--x  6 user_b  staff  192  6 Aug 17:21 process_a
+```
+
+The files in the subdirectory for a process, with the exception of a lock file, are accessible only by the owner of a process. The permissions
+for these files are `rwxr-xr-x` with the exception of the `attachNotificationSync` file, as shown in the following example.
+
+```
+~/tmp/.com_ibm_tools_attach/process_a $ ls -l
+total 4
+-rwxrw-rw-  1 user_b  staff  0  6 Aug 17:18 attachNotificationSync
+-rwxr-xr-x  1 user_b  staff  0  6 Aug 17:21 file_a
+-rwxr-xr-x  1 user_b  staff  0  6 Aug 17:21 file_b
+-rwxr-xr-x  1 user_b  staff  0  6 Aug 17:21 file_c
+```
+
+**Notes on z/OS:**
+
+- z/OS systems must also set POSIX permissions on files and cannot rely on RACF&reg; or system level security to protect applications.
+- To avoid z/OS console messages reporting security violations in `/tmp`, add a security exception or specify a different common subdirectory by setting the `com.ibm.tools.attach.directory` system property.
+
+## Configuring the Attach API
+
+A number of system properties are available to configure the Attach API when you start a Java application, as shown in the following table:
+
+| System property                                                                             |    Description                                                           |
+|---------------------------------------------------------------------------------------------|--------------------------------------------------------------------------|
+| [`-Dcom.ibm.tools.attach.directory=<directory_name>`](comibmtoolsattachdirectory.md)        | Specify a different directory in which to create `.com_ibm_tools_attach` |
+| [`-Dcom.ibm.tools.attach.displayName=<my_display_name>`](dcomibmtoolsattachdisplayname.md)  | Change the display name recorded by an agent                             |
+| [`-Dcom.ibm.tools.attach.id=<my_vm_ID>`](dcomibmtoolsattachid.md)                           | Change the VM identifier recorded by an agent                            |
+| [`-Dcom.ibm.tools.attach.timeout=<value_in_milliseconds>`](dcomibmtoolsattachtimeout.md)    | Change the connection timeout                                            |
+
```

Comment:
A few other system properties:
```
com.ibm.tools.attach.logging - Turn on tracing of attach API events. Valid values are "yes" and "no".
com.ibm.tools.attach.log.name - Override the default path for the log files generated by the "logging" option.
com.ibm.tools.attach.shutdown_timeout - Time delay before we give up trying to terminate the wait loop.
com.ibm.tools.attach.command_timeout - The socket timeout connecting to target VM.
```

Reply:
I assumed timeouts are in milliseconds. Can you provide a use case for the last 2 please? Are these just ways to shorten the `com.ibm.tools.attach.timeout` value when debugging?

Reply:
Yes, the timeouts are in milliseconds.
`com.ibm.tools.attach.timeout` is the timeout for initial connection to attach the target VM;
`com.ibm.tools.attach.command_timeout` is the timeout for sending a command to the target VM after initial attachment; 
`com.ibm.tools.attach.shutdown_timeout` is the timeout for the termination of the Attach API wait loop thread.
These timeouts are effective in different stages.

Reply:
The default value for `com.ibm.tools.attach.timeout` is `120000`;
The default value for `com.ibm.tools.attach.command_timeout` is `0`, i.e., no time out;
The default value for `com.ibm.tools.attach.timeout` is `10000`.